### PR TITLE
Rest API

### PR DESCRIPTION
**Pull request by andreasb242**
I added a REST API Server.
It allows to start / stop, query the state, query the config and allow to access the recorded files.

The implementation is minimalistic, and should probably be extended in the future.
But for now I think I added all features which I needed for now.

I'll test this Implementation in the future.

@woheller69 you might do a code review, you may find some issues.
I hope I didn't break anything.

I used swiftp as example for the network services, and swiftp was running on my Xiaomi device, where this app stopped unexpected. But swiftp was running all the time. I hope now that I also fixed this issue.

#14

### CODE REVIEW

**Comment by woheller69** [comment]
Will have a look next week. Does it work via ipv4 in a wifi only or also via ipv6 outdoors?

**Comment by andreasb242** [comment]
>Will have a look next week. Does it work via ipv4 in a wifi only or also via ipv6 outdoors?
I show the IPv4 address, and because there is at the moment absolutely no security (I started implementing password password protection, but did not yet finishing it yet) you shouldn't allow public access.

This can be extended later if needed, or maybe better, use a Proxy with security, e.g. setup an ngingx or apache with HTTPS, and password protection.

**Comment by woheller69** [comment]
Would it work if the phone with TimeLapseCamera provided a hotspot?

**Comment by andreasb242** [comment]
I didn't test it, I also don't have a SIM in the phone which I do timelapse.

You would like to access it remote?
I have to check if this is possible. If this is not possible, then a Portforwarding should be possible, e.g. over SSH.
But it can be made possible for sure in any way...

**Comment by andreasb242** [comment]
e.g. this would be a solution - and it should be secure:
https://github.com/regymm/termux-reverse-ssh

**Comment by woheller69** [comment]
I personally do not need that. But I guess there is the use case e.g. for a journalist where you place the cam somewhere and want to contol it. from a second device. 

(See here:
https://github.com/woheller69/TimeLapseCamera/issues/26)

The remote device might also provide the hotspot.

**Comment by andreasb242** [comment]
I think you linked the wrong ticket...

First I like to get this working stable, I do sometimes timelapse of growing plants, which takes weeks or month. I like to check the result without moving or touching the phone.
I record pictures, and the REST API Currently does only support pictures, no video (even I think it makes no sense to download an unfinished video, as you can read in #26, which you linked)

**Comment by woheller69** [comment]
I think remote start/stop should be provided for all kinds (image and video).
And if a video is finished, why not download it?
Otherwise I think this is too special...

**Comment by andreasb242** [comment]
>I think remote start/stop should be provided for all kinds (image and video).

Sure, there's no difference.

> And if a video is finished, why not download it?

Because I forgot tho add the MIME for Videos ;-)
I just added .jpg... But I anyway would add something, I'll add this.

**Comment by woheller69** [comment]
why do you add all the files in the .idea folder?

**Comment by andreasb242** [comment]
No special reason, I was not really sure if I should or not.
I can delete it, but then I also create a .gitignore file, which ignores it, so I'm not asked about it anymore.

I expect you would like to get the .gitignore?

**Comment by woheller69** [comment]
there is already a .gitignore
https://github.com/woheller69/TimeLapseCamera/blob/master/.gitignore which till now excluded the .idea files

just add the new filenames there

**Comment by andreasb242** [comment]
grafik
This is the template, no data filled in yet.

File: .gitignore [hunk 1]
```diff
@@ -1,15 +1,11 @@
 *.iml
 .gradle
 /local.properties
-/.idea/caches
```
Comment (andreasb242) [inline .gitignore:4]: The issue was, there were only some files excluded, not all. I changed this, so the whole folder is excluded now. I deleted it from this pull.

**Comment by woheller69** [comment]
no longer needed. The other PR was merged.

I added one commit to also update IpInformation after onResume()
Will publish new version V1.7 soon. Thanks!

**Pull request closed without merging**